// cloudmersive-nlp-api-client/src/index.js
/**
 * nlpapi
 * The powerful Natural Language Processing APIs let you perform part of speech tagging, entity identification, sentence parsing, and much more to help you understand the meaning of unstructured text.
 *
 * OpenAPI spec version: v1
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 *
 * Swagger Codegen version: unset
 *
 * Do not edit the class manually.
 *
 */

(function(factory) {
  if (typeof define === 'function' && define.amd) {
    // AMD. Register as an anonymous module.
    define(['ApiClient', 'model/CheckJsonResponse', 'model/CheckSentenceJsonResponse', 'model/CorrectJsonResponse', 'model/CorrectWordInSentenceJsonResponse', 'model/GetWordsJsonResponse', 'model/LanguageDetectionResponse', 'model/PosRequest', 'model/PosResponse', 'model/PosSentence', 'model/PosTaggedWord', 'model/WordPosition', 'api/ExtractEntitiesStringApi', 'api/LanguageDetectionApi', 'api/ParseStringApi', 'api/PosTaggerJsonApi', 'api/PosTaggerStringApi', 'api/SentencesApi', 'api/SpellCheckApi', 'api/WordsApi'], factory);
  } else if (typeof module === 'object' && module.exports) {
    // CommonJS-like environments that support module.exports, like Node.
    module.exports = factory(require('./ApiClient'), require('./model/CheckJsonResponse'), require('./model/CheckSentenceJsonResponse'), require('./model/CorrectJsonResponse'), require('./model/CorrectWordInSentenceJsonResponse'), require('./model/GetWordsJsonResponse'), require('./model/LanguageDetectionResponse'), require('./model/PosRequest'), require('./model/PosResponse'), require('./model/PosSentence'), require('./model/PosTaggedWord'), require('./model/WordPosition'), require('./api/ExtractEntitiesStringApi'), require('./api/LanguageDetectionApi'), require('./api/ParseStringApi'), require('./api/PosTaggerJsonApi'), require('./api/PosTaggerStringApi'), require('./api/SentencesApi'), require('./api/SpellCheckApi'), require('./api/WordsApi'));
  }
}(function(ApiClient, CheckJsonResponse, CheckSentenceJsonResponse, CorrectJsonResponse, CorrectWordInSentenceJsonResponse, GetWordsJsonResponse, LanguageDetectionResponse, PosRequest, PosResponse, PosSentence, PosTaggedWord, WordPosition, ExtractEntitiesStringApi, LanguageDetectionApi, ParseStringApi, PosTaggerJsonApi, PosTaggerStringApi, SentencesApi, SpellCheckApi, WordsApi) {
  'use strict';

  /**
   * The_powerful_Natural_Language_Processing_APIs_let_you_perform_part_of_speech_tagging_entity_identification_sentence_parsing_and_much_more_to_help_you_understand_the_meaning_of_unstructured_text_.<br>
   * The <code>index</code> module provides access to constructors for all the classes which comprise the public API.
   * <p>
   * An AMD (recommended!) or CommonJS application will generally do something equivalent to the following:
   * <pre>
   * var CloudmersiveNlpApiClient = require('index'); // See note below*.
   * var xxxSvc = new CloudmersiveNlpApiClient.XxxApi(); // Allocate the API class we're going to use.
   * var yyyModel = new CloudmersiveNlpApiClient.Yyy(); // Construct a model instance.
   * yyyModel.someProperty = 'someValue';
   * ...
   * var zzz = xxxSvc.doSomething(yyyModel); // Invoke the service.
   * ...
   * </pre>
   * <em>*NOTE: For a top-level AMD script, use require(['index'], function(){...})
   * and put the application logic within the callback function.</em>
   * </p>
   * <p>
   * A non-AMD browser application (discouraged) might do something like this:
   * <pre>
   * var xxxSvc = new CloudmersiveNlpApiClient.XxxApi(); // Allocate the API class we're going to use.
   * var yyy = new CloudmersiveNlpApiClient.Yyy(); // Construct a model instance.
   * yyyModel.someProperty = 'someValue';
   * ...
   * var zzz = xxxSvc.doSomething(yyyModel); // Invoke the service.
   * ...
   * </pre>
   * </p>
   * @module index
   * @version 1.1.2
   */
  var exports = {
    /**
     * The ApiClient constructor.
     * @property {module:ApiClient}
     */
    ApiClient: ApiClient,
    /**
     * The CheckJsonResponse model constructor.
     * @property {module:model/CheckJsonResponse}
     */
    CheckJsonResponse: CheckJsonResponse,
    /**
     * The CheckSentenceJsonResponse model constructor.
     * @property {module:model/CheckSentenceJsonResponse}
     */
    CheckSentenceJsonResponse: CheckSentenceJsonResponse,
    /**
     * The CorrectJsonResponse model constructor.
     * @property {module:model/CorrectJsonResponse}
     */
    CorrectJsonResponse: CorrectJsonResponse,
    /**
     * The CorrectWordInSentenceJsonResponse model constructor.
     * @property {module:model/CorrectWordInSentenceJsonResponse}
     */
    CorrectWordInSentenceJsonResponse: CorrectWordInSentenceJsonResponse,
    /**
     * The GetWordsJsonResponse model constructor.
     * @property {module:model/GetWordsJsonResponse}
     */
    GetWordsJsonResponse: GetWordsJsonResponse,
    /**
     * The LanguageDetectionResponse model constructor.
     * @property {module:model/LanguageDetectionResponse}
     */
    LanguageDetectionResponse: LanguageDetectionResponse,
    /**
     * The PosRequest model constructor.
     * @property {module:model/PosRequest}
     */
    PosRequest: PosRequest,
    /**
     * The PosResponse model constructor.
     * @property {module:model/PosResponse}
     */
    PosResponse: PosResponse,
    /**
     * The PosSentence model constructor.
     * @property {module:model/PosSentence}
     */
    PosSentence: PosSentence,
    /**
     * The PosTaggedWord model constructor.
     * @property {module:model/PosTaggedWord}
     */
    PosTaggedWord: PosTaggedWord,
    /**
     * The WordPosition model constructor.
     * @property {module:model/WordPosition}
     */
    WordPosition: WordPosition,
    /**
     * The ExtractEntitiesStringApi service constructor.
     * @property {module:api/ExtractEntitiesStringApi}
     */
    ExtractEntitiesStringApi: ExtractEntitiesStringApi,
    /**
     * The LanguageDetectionApi service constructor.
     * @property {module:api/LanguageDetectionApi}
     */
    LanguageDetectionApi: LanguageDetectionApi,
    /**
     * The ParseStringApi service constructor.
     * @property {module:api/ParseStringApi}
     */
    ParseStringApi: ParseStringApi,
    /**
     * The PosTaggerJsonApi service constructor.
     * @property {module:api/PosTaggerJsonApi}
     */
    PosTaggerJsonApi: PosTaggerJsonApi,
    /**
     * The PosTaggerStringApi service constructor.
     * @property {module:api/PosTaggerStringApi}
     */
    PosTaggerStringApi: PosTaggerStringApi,
    /**
     * The SentencesApi service constructor.
     * @property {module:api/SentencesApi}
     */
    SentencesApi: SentencesApi,
    /**
     * The SpellCheckApi service constructor.
     * @property {module:api/SpellCheckApi}
     */
    SpellCheckApi: SpellCheckApi,
    /**
     * The WordsApi service constructor.
     * @property {module:api/WordsApi}
     */
    WordsApi: WordsApi
  };

  return exports;
}));
