// cloudmersive-validate-api-client/src/index.js
/**
 * validateapi
 * The validation APIs help you validate data. Check if an E-mail address is real. Check if a domain is real. Check up on an IP address, and even where it is located. All this and much more is available in the validation API.
 *
 * OpenAPI spec version: v1
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 *
 * Swagger Codegen version: unset
 *
 * Do not edit the class manually.
 *
 */

(function(factory) {
  if (typeof define === 'function' && define.amd) {
    // AMD. Register as an anonymous module.
    define(['ApiClient', 'model/AddressGetServersResponse', 'model/AddressVerifySyntaxOnlyResponse', 'model/CheckResponse', 'model/FullEmailValidationResponse', 'model/GeolocateResponse', 'model/PhoneNumberValidateRequest', 'model/PhoneNumberValidationResponse', 'model/VatLookupRequest', 'model/VatLookupResponse', 'model/WhoisResponse', 'api/DomainApi', 'api/EmailApi', 'api/IPAddressApi', 'api/PhoneNumberApi', 'api/VatApi'], factory);
  } else if (typeof module === 'object' && module.exports) {
    // CommonJS-like environments that support module.exports, like Node.
    module.exports = factory(require('./ApiClient'), require('./model/AddressGetServersResponse'), require('./model/AddressVerifySyntaxOnlyResponse'), require('./model/CheckResponse'), require('./model/FullEmailValidationResponse'), require('./model/GeolocateResponse'), require('./model/PhoneNumberValidateRequest'), require('./model/PhoneNumberValidationResponse'), require('./model/VatLookupRequest'), require('./model/VatLookupResponse'), require('./model/WhoisResponse'), require('./api/DomainApi'), require('./api/EmailApi'), require('./api/IPAddressApi'), require('./api/PhoneNumberApi'), require('./api/VatApi'));
  }
}(function(ApiClient, AddressGetServersResponse, AddressVerifySyntaxOnlyResponse, CheckResponse, FullEmailValidationResponse, GeolocateResponse, PhoneNumberValidateRequest, PhoneNumberValidationResponse, VatLookupRequest, VatLookupResponse, WhoisResponse, DomainApi, EmailApi, IPAddressApi, PhoneNumberApi, VatApi) {
  'use strict';

  /**
   * The_validation_APIs_help_you_validate_data__Check_if_an_E_mail_address_is_real__Check_if_a_domain_is_real__Check_up_on_an_IP_address_and_even_where_it_is_located__All_this_and_much_more_is_available_in_the_validation_API_.<br>
   * The <code>index</code> module provides access to constructors for all the classes which comprise the public API.
   * <p>
   * An AMD (recommended!) or CommonJS application will generally do something equivalent to the following:
   * <pre>
   * var CloudmersiveValidateApiClient = require('index'); // See note below*.
   * var xxxSvc = new CloudmersiveValidateApiClient.XxxApi(); // Allocate the API class we're going to use.
   * var yyyModel = new CloudmersiveValidateApiClient.Yyy(); // Construct a model instance.
   * yyyModel.someProperty = 'someValue';
   * ...
   * var zzz = xxxSvc.doSomething(yyyModel); // Invoke the service.
   * ...
   * </pre>
   * <em>*NOTE: For a top-level AMD script, use require(['index'], function(){...})
   * and put the application logic within the callback function.</em>
   * </p>
   * <p>
   * A non-AMD browser application (discouraged) might do something like this:
   * <pre>
   * var xxxSvc = new CloudmersiveValidateApiClient.XxxApi(); // Allocate the API class we're going to use.
   * var yyy = new CloudmersiveValidateApiClient.Yyy(); // Construct a model instance.
   * yyyModel.someProperty = 'someValue';
   * ...
   * var zzz = xxxSvc.doSomething(yyyModel); // Invoke the service.
   * ...
   * </pre>
   * </p>
   * @module index
   * @version 1.1.2
   */
  var exports = {
    /**
     * The ApiClient constructor.
     * @property {module:ApiClient}
     */
    ApiClient: ApiClient,
    /**
     * The AddressGetServersResponse model constructor.
     * @property {module:model/AddressGetServersResponse}
     */
    AddressGetServersResponse: AddressGetServersResponse,
    /**
     * The AddressVerifySyntaxOnlyResponse model constructor.
     * @property {module:model/AddressVerifySyntaxOnlyResponse}
     */
    AddressVerifySyntaxOnlyResponse: AddressVerifySyntaxOnlyResponse,
    /**
     * The CheckResponse model constructor.
     * @property {module:model/CheckResponse}
     */
    CheckResponse: CheckResponse,
    /**
     * The FullEmailValidationResponse model constructor.
     * @property {module:model/FullEmailValidationResponse}
     */
    FullEmailValidationResponse: FullEmailValidationResponse,
    /**
     * The GeolocateResponse model constructor.
     * @property {module:model/GeolocateResponse}
     */
    GeolocateResponse: GeolocateResponse,
    /**
     * The PhoneNumberValidateRequest model constructor.
     * @property {module:model/PhoneNumberValidateRequest}
     */
    PhoneNumberValidateRequest: PhoneNumberValidateRequest,
    /**
     * The PhoneNumberValidationResponse model constructor.
     * @property {module:model/PhoneNumberValidationResponse}
     */
    PhoneNumberValidationResponse: PhoneNumberValidationResponse,
    /**
     * The VatLookupRequest model constructor.
     * @property {module:model/VatLookupRequest}
     */
    VatLookupRequest: VatLookupRequest,
    /**
     * The VatLookupResponse model constructor.
     * @property {module:model/VatLookupResponse}
     */
    VatLookupResponse: VatLookupResponse,
    /**
     * The WhoisResponse model constructor.
     * @property {module:model/WhoisResponse}
     */
    WhoisResponse: WhoisResponse,
    /**
     * The DomainApi service constructor.
     * @property {module:api/DomainApi}
     */
    DomainApi: DomainApi,
    /**
     * The EmailApi service constructor.
     * @property {module:api/EmailApi}
     */
    EmailApi: EmailApi,
    /**
     * The IPAddressApi service constructor.
     * @property {module:api/IPAddressApi}
     */
    IPAddressApi: IPAddressApi,
    /**
     * The PhoneNumberApi service constructor.
     * @property {module:api/PhoneNumberApi}
     */
    PhoneNumberApi: PhoneNumberApi,
    /**
     * The VatApi service constructor.
     * @property {module:api/VatApi}
     */
    VatApi: VatApi
  };

  return exports;
}));
